docs/src/developer/interfaces.md

```diff
@@ -0,0 +1,183 @@
+
+* Iteration : iterators are currently provided for multivariate polynomials to
+iterate over the coefficients, terms and monomials. Nemo matrices can also be
+iterated over. Iteration proceeds down each column in turn. One can also
+iterate over all permutations and partitions. Finally, all finite field types
+can be iterated over.
+
+* Views : because C libraries cannot be expected to implement the full range
+of Julia view types, views of matrices in Nemo can only be constructed for
+submatrices consisting of contiguous blocks in the original matrix.
+
+* map and similar : we implement the map and similar interfaces with the
+caveat that we generally use parent objects where Julia would use types. See
+the specific documentation for the module of interest to see details.
+
+* zero and one : these are implemented for parent types, which is not what
+Julia typically expects. Exceptions include the Flint `fmpz` and `fmpq` types,
+as their parents are not parameterised, which makes it possible to implement
+these functions for the types as well as the parents.
+
+* rand : we have a Nemo specific `rand` interface, which passes the tail of
+a given `rand` invocation to the `rand` function for the base ring, e.g. to
+create random matrix elements or polynomial coefficients and so on. In addition
+to this custom `rand` interface, we also support much of the Julia `rand`
+interface, with the usual caveat that we use parent objects instead of types
+where necessary.
+
+* serialisation : unfortunately this is currently NOT implemented by Nemo, but
+we would certainly like to see that done in the future. It's not automatic
+because of the C objects that underly many of our constructions.
+
+* Number : Nemo number types do NOT belong to Julia's `Number` hierarchy, as
+we must make all our ring element types belong to our `RingElem` abstract
+type. To make some Julia `Number` types cooperate with Nemo, we define the
+unions `RingElement` and `FieldElement` which include some Julia types, such
+as `BigInt` and `Rational{BigInt}`, etc. Note that fixed precision integer
+types cannot be expected to be well-behaved when they overflow. We recommend
+using Nemo integer types if one wants good performance for small machine
+word sized integers, but no overflow when the integer becomes large (Nemo
+integers are based on Flint's multiprecision `fmpz` type).
+
+* hash : we implement hash functions for all major element types in Nemo.
+
+* getindex/setindex! : we implement these to access elements of Nemo matrices,
+however see the note below on row major representation. In addition, we allow
+creation of matrices using the notation `R[a b; c d]` etc. This is done by
+overloading `getindex` for the parent object `R` instead of a type as Julia
+would normally expect. Note that we also provide this overload for the types
+`fmpz` and `fmpq` as these types are not parameterised and so matrices can
+be constructed using the type only.
```

Comment:
"We" don't provide overloads, e.g. `fmpz[1 2]` uses the `Base` definition and creates a julia `Matrix`, not a `Nemo` matrix, so this syntax just follows normal rules (and it should stay this way). So this note should be removed or corrected.

Reply:
Right, of course. I have fixed this and the `typed_hvcat` thing.